### ISSUE TITLE
Cronjob UI tab not namespaced

### ISSUE THREAD

**Issue by dal13002** [headlamp-k8s/headlamp#2687]
## Describe the bug
As of [this](https://github.com/headlamp-k8s/headlamp/pull/2634) pull request, we can now select a namespace from the filter search bar on almost any page (pods, deployments, secrets, etc..) and all api calls are being scoped to that namespace. This is great since we don't have to modify / mange "accessible namespaces" for non-admin users. What I've noticed is this feature is not happening in the "cronjobs" tab, and I am still getting errors trying to access cronjobs when I filter to a certain namespace.

## To Reproduce
Steps to reproduce the bug:
1. You have full access to "test" namespace and list all namespaces within the cluster
2. There is nothing configured in "accessible namespaces" setting
3. Create a cronjob in "test" namespace (ie use kubectl)
4. Go to cronjob tab on headlamp -> add "test" to the namespace filter
5. The UI seems to just load forever

## Environment (please provide info about your environment):
 - Installation type: In cluster using helm
 - Headlamp Version: v0.27.0

## Are you able to fix this issue?
No

## Additional Context
N/A

**Comment by skoeva** [comment]
Hi, are you still experiencing this issue? If so, would you be able to share more info about it (i.e. how long the table takes to load, how many cron jobs you have configured, or anything else that might be helpful)

**Comment by dal13002** [comment]
Hi,
Yes I still see this issue. I have only 1 cronjob and it will load forever (the load icon never goes away). Basically:
- if I give myself cluster-admin access, page will load immediately and I will see the 1 cronjob
- if I give myself admin to just the "test" namespace (where the one cronjob lives), and list all namespaces permission- I can select the "test" namespace from the filter menu. But it will load forever. If I look at the network tab, I see 403 forbidden while calling `api/batch/v1/cronjobs`. I think this api call should be namespaced but it is not- for example, when I select the "test" namespace it should be calling something like `api/batch/v1/namespace/test/cronjobs` (I don't know the internal endpoints but something like that so it only targets that namespace the same way "pods" work)

**Comment by adwait-godbole** [comment]
Hi @dal13002, Hi @skoeva. 

I think I just found out what the underlying issue might be over here. Nothing is wrong with Headlamp's code. 

@dal13002 I replicated your steps and found out that if you look closely at the 403 Forbidden Error Response Message on the CronJobs List Page, it should say something like below.

![Image](https://github.com/user-attachments/assets/862e7a89-6630-41c3-9379-2e26c1ffe97b)

The problem is although your ServiceAccount has admin access to "test" namespace along with access to "list" other namespaces, when it comes to listing Kubernetes resources under the "test" namespace, by default it is only allowed to list resources which are part of the Core group. Some examples being: Pods, ResourceQuotas, Secrets, etc. 

You can verify this by running the below commands: 

(Side Note: during replicating your steps on my local, i had modified permissions of ServiceAccount `headlamp-admin` itself to only have admin access to "test" namespace and ability to list namespaces)

```bash
adwait@vostro-3500:/media/adwait/DATA1/LFX/headlamp$ kubectl auth can-i list pods -n test --as=system:serviceaccount:kube-system:headlamp-admin
yes
```

```bash
adwait@vostro-3500:/media/adwait/DATA1/LFX/headlamp$ kubectl auth can-i list cronjobs -n test --as=system:serviceaccount:kube-system:headlamp-admin
no
```

In order to fix this you can take the below steps:

![Image](https://github.com/user-attachments/assets/3708af6b-d904-4d80-bbcb-d2bea6cb311a)

I hope this resolves your query :). Thanks.

**Comment by dal13002** [comment]
@adwait-godbole I don't agree with the way you tested this. When I say "service account has admin to  he namespace test", I am not just talking about having access to the "core" api group, I mean having access to all resources / api groups in that namespace. Obviously, if you don't grant a user / service account access to cronjobs in a namespace, he will not be able to see cronjob via kubectl. You should give the user access to cronjobs in that specific namespace (ie `kubectl auth can-i list cronjobs -n test --as=system:serviceaccount:kube-system:headlamp-admin` and `kubectl get cronjobs -n test` return successfully), and you will see that the cronjob tab in Headlamp still get stuck in loading forever.

**Comment by dal13002** [comment]
@adwait-godbole I don't think the problem was that I was only granting permission to the "core" api group. When I said "admin to the namespace", I meant add permission to all resources / api groups to that namespace. I think this issue is fixed in 0.28 release, and I would guess you are running that version but correct me if I'm wrong. Closing this ticket since I don't see the issue anymore

**Comment by adwait-godbole** [comment]
> I think this issue is fixed in 0.28 release, and I would guess you are running that version but correct me if I'm wrong

Yes @dal13002. I was testing all of this on the latest version i.e. 0.28. 

But my bad for assuming that you hadn’t configured access / mistakenly forgot to add it for other than the Core API group. 

I just thought maybe that might be the issue behind you seeing the 403 error. But looks like 0.27 had this bug and yes, it no more exists in 0.28. Glad to see that this has been resolved.

I learnt a lot of new things along the way while trying to debug this issue.

Thanks @dal13002.

**Comment by dal13002** [comment]
@adwait-godbole thanks for looking into it. I appreciate you taking the time to debug and post your finding to help the community
